Specify Python version

[calculator.py](https://github.com/Cambridge-ICCS/ss23-git/blob/main/rpn-calc/src/calculator.py) uses the `match` construct which was added in Python 3.10. It would be good to mention that this is therefore the minimum required version of python in the README.

Reproducer in Python 3.9

```
================================================= test session starts ==================================================
platform linux -- Python 3.9.12, pytest-7.4.0, pluggy-1.2.0
rootdir: /home/cje57/ICCS/SS23/ss23-git/rpn-calc/src
collected 0 items / 1 error                                                                                            

======================================================== ERRORS ========================================================
_______________________________________ ERROR collecting tests/test_calculate.py _______________________________________
../../venv/lib/python3.9/site-packages/_pytest/python.py:617: in _importtestmodule
    mod = import_path(self.path, mode=importmode, root=self.config.rootpath)
../../venv/lib/python3.9/site-packages/_pytest/pathlib.py:565: in import_path
    importlib.import_module(module_name)
/home/cje57/anaconda3/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1030: in _gcd_import
    ???
<frozen importlib._bootstrap>:1007: in _find_and_load
    ???
<frozen importlib._bootstrap>:986: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:680: in _load_unlocked
    ???
../../venv/lib/python3.9/site-packages/_pytest/assertion/rewrite.py:178: in exec_module
    exec(co, module.__dict__)
tests/test_calculate.py:1: in <module>
    from rpn_calc import main
rpn_calc.py:7: in <module>
    from calculator import Calculator
E     File "/home/cje57/ICCS/SS23/ss23-git/rpn-calc/src/calculator.py", line 21
E       match token:
E             ^
E   SyntaxError: invalid syntax
=============================================== short test summary info ================================================
ERROR tests/test_calculate.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
=================================================== 1 error in 0.07s ===================================================
```

--- a/README.md
+++ b/README.md
@@ -55,3 +55,5 @@ pip install pytest
 cd rpn-calc/src
 pytest
 ```
+
+The minimum required version of Python is 3.10.

--- a/rpn-calc/src/rpn_calc.py
+++ b/rpn-calc/src/rpn_calc.py
@@ -10,6 +10,9 @@ from calculator import Calculator
 if TYPE_CHECKING:
     from collections.abc import Iterable
 
+MIN_PYTHON = (3, 10)
+if sys.version_info < MIN_PYTHON:
+    sys.exit("Python %s.%s or later is required.\n" % MIN_PYTHON)
 
 def main(commands: "Iterable" = sys.stdin):
     '''

